tests/test_stl_binders.cpp

```diff
@@ -129,6 +163,12 @@ TEST_SUBMODULE(stl_binders, m) {
             m, "VectorUndeclStruct", py::buffer_protocol());
     });
 
+    // Bind recursive container types
+    py::bind_vector<RecursiveVector>(m, "RecursiveVector");
```

Comment:
Do these tests cover the `detail:is_move_constructable` templates too?

Reply:
I have added static testing for the pybind11 type traits now (and coincidentally, uncovered a slight bug in `pybind11::detail::is_copy_assignable`)